client/my-sites/stats/hooks/use-should-gate-stats.ts

```diff
@@ -84,7 +95,9 @@ export const shouldGateStats = ( state: object, siteId: number | null, statType:
 	if ( jetpackSite && ! atomicSite ) {
 		// TODO: Determine more paywall segments and granular control for paid stats.
 		if ( restrictDashboard && isNewSite && ! hasAnyStatsPlan ) {
```

Comment:
Should target only commercial sites here?

Reply:
Yes, let's clarify the logic in the follow-up PR later.